ARE/middleware/src/main/java/eu/asterics/mw/are/BundleManager.java

```diff
@@ -132,16 +144,173 @@ public BundleManager(ModelValidator modelValidator) {
+        File pluginJarsHashFile=ResourceRegistry.toFile(pluginJarsHashFileURI);
+        if(!pluginJarsHashFile.exists()) {
+            logger.warning("pluginJars hash file does not exist --> bundle change: true");
+            return true;
+        }
+        try(BufferedReader reader=new BufferedReader(new FileReader(pluginJarsHashFile))) {
+            String storedMD5Sum=reader.readLine();
+            if(md5Sum.equals(storedMD5Sum)) {
+                logger.fine("bundle change: false");
+                return false;
+            }
+        }
+        
+        logger.fine("bundle change: true");
+        return true;
+    }
+
+    /**
+     * Stores the given MD5 hash hex string to a file.
+     * @param md5Sum
+     * @throws URISyntaxException
+     * @throws IOException
+     */
+    private void storeMD5Sum(String md5Sum) throws URISyntaxException, IOException {
+        logger.fine("Storing MD5 sum...");
+        URI pluginJarsHashFileURI=ResourceRegistry.getInstance().getResource(CACHING_PLUGIN_JARS_HASH_TXT, RES_TYPE.TMP);
+        File pluginJarsHashFile=ResourceRegistry.toFile(pluginJarsHashFileURI);
+        File parentFile=pluginJarsHashFile.getParentFile();
+        
+        if(!parentFile.exists()) {
+            if(!parentFile.mkdirs()) {
+                throw new IOException("Could not create parent directories for MD5 hash file: "+pluginJarsHashFile);
+            }
+        }
+        try(FileWriter writer=new FileWriter(pluginJarsHashFile)) {
+            writer.write(md5Sum.trim());
+        }
+    }
+
+    /**
+     * This method generates all needed cachde files.
+     * @throws MalformedURLException
+     * @throws IOException
+     * @throws ParseException
+     * @throws URISyntaxException
+     */
+    private void generateCacheFiles() throws MalformedURLException, IOException, ParseException, URISyntaxException {        
+        logger.fine("Generating cache files/help files");
+        long startTime=System.currentTimeMillis();       
+        //Before we can start, let's ensure the generation of cached files
+        generateLoaderComponentListAndBundleDescriptorCacheFiles();
+        logger.fine(MessageFormat.format("Generated cache files in {0} ms",(System.currentTimeMillis()-startTime)));
+    }
+
+    /**
+     * Generates an MD5 hex string of the bytes of all plugin jars.
+     * A plugin jar is defined as an osgi bundle containing a bundle_descriptor.xml file in the root folder of the bundle.
+     * @return
+     * @throws NoSuchAlgorithmException
+     * @throws MalformedURLException
+     * @throws IOException
+     */
+    private String generateMD5Sum() throws NoSuchAlgorithmException, MalformedURLException, IOException {
+        logger.fine("Generating MD5 sum of bundle jars....");
+        long timeStart=System.currentTimeMillis();
+        MessageDigest MDInstance=MessageDigest.getInstance("MD5");
```

Comment:
`MDInstance` is a strange name, I think it should start with a small letter, like all variables. So e.g. `mdInstance`?

Reply:
ok